services/softlayer_dns_domain_record_test.go

```diff
@@ -0,0 +1,133 @@
+package services_test
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	slclientfakes "github.com/maximilien/softlayer-go/client/fakes"
+	datatypes "github.com/maximilien/softlayer-go/data_types"
+	softlayer "github.com/maximilien/softlayer-go/softlayer"
+	testhelpers "github.com/maximilien/softlayer-go/test_helpers"
+)
+
+var _ = Describe("SoftLayer_Dns_Domain_Record", func() {
+	var (
+		username, apiKey string
+
+		fakeClient *slclientfakes.FakeSoftLayerClient
+
+		dnsDomainRecordService softlayer.SoftLayer_Dns_Domain_Record_Service
+		err                    error
+	)
+
+	BeforeEach(func() {
+		username = os.Getenv("SL_USERNAME")
+		Expect(username).ToNot(Equal(""))
+
+		apiKey = os.Getenv("SL_API_KEY")
+		Expect(apiKey).ToNot(Equal(""))
+
+		fakeClient = slclientfakes.NewFakeSoftLayerClient(username, apiKey)
+		Expect(fakeClient).ToNot(BeNil())
+
+		dnsDomainRecordService, err = fakeClient.GetSoftLayer_Dns_Domain_Record_Service()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(dnsDomainRecordService).ToNot(BeNil())
+	})
+
+	Context("#GetName", func() {
+		It("returns the name for the service", func() {
+			name := dnsDomainRecordService.GetName()
+			Expect(name).To(Equal("SoftLayer_Dns_Domain_ResourceRecord"))
+		})
+	})
+
+	Context("#CreateObject", func() {
```

Comment:
Adding a few cases when each method fails would be cool. Don't over do it but some of the common failure cases would be great. You can control the fake client to return error and make sure it's propagated.